services/ui-src/src/components/overlays/EntityDetailsMultiformOverlay.tsx

```diff
@@ -111,13 +114,16 @@ export const EntityDetailsMultiformOverlay = ({
         const exceptionKeys: string[] = [];
         const nonComplianceKeys: string[] = [];
 
-        Object.keys(selectedEntity || {}).forEach((key) => {
-          if (key.startsWith(standardKeyPrefix)) {
+        // look through existing and new keys, because we may need to delete either
+        const combinedData = { ...selectedEntity, ...enteredData };
```

Comment:
since we might delete keys, and they might live in either object, we need to check both for the applicable keys.

previously, we only looked at `selectedEntity` keys, but this created a problem where if we had a new key in the enteredData that should be deleted, we wouldn't find it. (this could happen if you select a checkbox in the form and then unselect it)